src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -57,12 +58,14 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
             var addImportService = document.GetLanguageService<IAddImportFeatureService>();
 
             var solution = document.Project.Solution;
-            var options = solution.Options;
 
-            var searchReferenceAssemblies = options.GetOption(SymbolSearchOptions.SuggestForTypesInReferenceAssemblies, document.Project.Language);
-            var searchNuGetPackages = options.GetOption(SymbolSearchOptions.SuggestForTypesInNuGetPackages, document.Project.Language);
+            var searchNuGetPackages = solution.Options.GetOption(SymbolSearchOptions.SuggestForTypesInNuGetPackages, document.Project.Language);
 
-            var symbolSearchService = searchReferenceAssemblies || searchNuGetPackages
+            var options = new AddImportOptions(
+                context.Options.SearchReferenceAssemblies,
+                context.Options.HideAdvancedMembers);
+
+            var symbolSearchService = options.SearchReferenceAssemblies || searchNuGetPackages
```

Comment:
consider leaving the original code here, and just moving the options creation below to where it is first needed.